### PR TITLE
Backend tests

### PR DESCRIPTION
**Pull request by paulgoujon**
- Restructured the test division : each model in a file, each form in a file, each view in a file ...
- Searchform test provided : PLEASE READ AND COMMENT IT

### CODE REVIEW

File: bagtrekkin/tests/test_searchForm.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        self.client.login(username='capflam', password='123')
+
+    def test_init_no_data(self):
+        """Init should be successful without provided data"""
+        form = SearchForm()
+        self.assertTrue(isinstance(form, SearchForm))
+
+    def test_init_pnr_data(self):
+        """Init should be successful with pnr provided data"""
+        form = SearchForm({'pnr': 'YSVI82'})
```
Comment (roddehugo) [inline bagtrekkin/tests/test_searchForm.py]: Possible to factors both inits in one method. 
See https://github.com/goujonpa/bagtrekkin/blob/roddehugo_tastypie_tests/bagtrekkin/api/tests.py#L32-L36
Create a method with a name not starting with test which takes to arguments `key` and `value`
Then both init tests will call that method ;)

File: bagtrekkin/tests/test_searchForm.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.assertTrue(isinstance(form, SearchForm))
+        self.assertTrue(form.is_valid())
+        self.assertEqual(form.cleaned_data['pnr'], 'YSVI82')
+
+    def test_init_rfid_data(self):
+        """Init should be successful with rfid provided data"""
+        form = SearchForm({'material_number': 'FFFFFFFF'})
```
Comment (roddehugo) [inline bagtrekkin/tests/test_searchForm.py]: See comment above

File: bagtrekkin/tests/test_searchForm.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    def test_init_two_data_not_valid(self):
+        """Form with pnr AND material_number shouldn't be valid"""
+        form = SearchForm({'pnr': 'YSVI82', 'material_number': 'FFFFFFFF'})
+        self.assertTrue(isinstance(form, SearchForm))
+        self.assertFalse(form.is_valid())
+
+    def test_search_good_pnr(self):
+        """Found objects should correspond to what we searched"""
+        form = SearchForm({'pnr': 'ABC123'})
```
Comment (roddehugo) [inline bagtrekkin/tests/test_searchForm.py]: Those three lines are often repeated, you can use the method described above to get a factorization

File: bagtrekkin/tests/test_searchForm.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        self.assertTrue(isinstance(form, SearchForm))
+        self.assertTrue(form.is_valid())
```
Comment (roddehugo) [inline bagtrekkin/tests/test_searchForm.py]: Those three lines above are often repeated, you can use the method described above to get a factorization

File: bagtrekkin/views.py [hunk 5]
```diff
@@ -40,31 +40,24 @@ def signup(request):
...
-            try:
+        context = {
+            'search_form': search_form
+        }
+        try:
+            if search_form.is_valid():
                 passenger, luggages, logs = search_form.search()
-                context = {
-                    'search_form': search_form,
+                context.update({
```
Comment (roddehugo) [inline bagtrekkin/views.py:49]: Yeah! :+1:

**Comment by paulgoujon** [comment]
- Code modified as advised
- deleted tests.py (not useful anymore)

kiss <3

**Pull request merged**